src/test/java/com/basho/riak/client/api/commands/itest/ITestDatatype.java

```diff
@@ -192,37 +190,92 @@ public void testNotFoundHyperLogLog() throws ExecutionException, InterruptedExce
         assertTrue(updatedSet.contains(BinaryValue.create("user2")));
         assertTrue(updatedSet.contains(BinaryValue.create("user3")));
-        assertFalse(updateResponse.hasContext());
+        assertFalse(updatedSet.contains(BinaryValue.create("user4")));
+
+        final FetchSet.Response removeItemResponse = client.execute(fetchSet);
+        Context ctx = removeItemResponse.getContext();
+        SetUpdate suRemoveItem = new SetUpdate().remove("user2");
+        UpdateSet updateRemove = new UpdateSet.Builder(location, suRemoveItem).withContext(ctx).build();
+        client.execute(updateRemove);
+
+        final FetchSet.Response removedResponse = client.execute(fetchSet);
+        Set<BinaryValue> removedItemSet = removedResponse.getDatatype().view();
+        assertFalse(removedItemSet.isEmpty());
+        assertTrue(removedItemSet.contains(BinaryValue.create("user1")));
+        assertFalse(removedItemSet.contains(BinaryValue.create("user2")));
+        assertTrue(removedItemSet.contains(BinaryValue.create("user3")));
+        assertFalse(removedItemSet.contains(BinaryValue.create("user4")));
+    }
+
+    @Test
+    public void testGSet() throws ExecutionException, InterruptedException {
+        Assume.assumeTrue(testGSetDataType);
+        resetAndEmptyBucket(uniqueUsersGSet);
+
+        final Location location = new Location(uniqueUsersGSet, "users-" + new Random().nextLong());
+
+        FetchSet fetchSet = new FetchSet.Builder(location).build();
+        final FetchSet.Response initialFetchResponse = client.execute(fetchSet);
+
+        final RiakSet initialSet = initialFetchResponse.getDatatype();
+        assertTrue(initialSet.view().isEmpty());
 
-        final FetchSet.Response loadedFetchResponse = client.execute(fetchSet);
+        GSetUpdate gsu = new GSetUpdate()
+                .add("user1")
+                .add("user2")
+                .add("user3");
+        UpdateSet us = new UpdateSet.Builder(location, gsu).build();
+        client.execute(us);
 
-        final Set<BinaryValue> loadedSet = loadedFetchResponse.getDatatype().view();
+        final FetchSet.Response newItemResponse = client.execute(fetchSet);
+        final Set<BinaryValue> updatedSet = newItemResponse.getDatatype().view();
+        assertFalse(updatedSet.isEmpty());
+        assertTrue(updatedSet.contains(BinaryValue.create("user1")));
+        assertTrue(updatedSet.contains(BinaryValue.create("user2")));
+        assertTrue(updatedSet.contains(BinaryValue.create("user3")));
+        assertFalse(updatedSet.contains(BinaryValue.create("user4")));
+
+        final FetchSet.Response removeItemResponse = client.execute(fetchSet);
+        Context ctx = removeItemResponse.getContext();
+        try {
+            SetUpdate suRemoveItem = new SetUpdate().remove("user2");
+            UpdateSet updateRemove = new UpdateSet.Builder(location, suRemoveItem).withContext(ctx).build();
+            client.execute(updateRemove);
```

Comment:
After this, there should be an Assert that always fails as it should never get to the assert line due to an exception being thrown.